include/posix/time.h

```diff
@@ -10,7 +10,25 @@
 extern "C" {
 #endif
 
+#ifdef CONFIG_NEWLIB_LIBC
+/* Kludge to support outdated newlib version as used in SDK 0.10 for Xtensa */
```

Comment:
If this is just for XTensa, why not ifndef under that arch too, so other arches for sure aren't negatively impacted by unexpected changes here?

Reply:
> If this is just for XTensa,

And what if this not only by Xtensa, and later we have another FooBarArch with the same issue? This kludge tests for the actual indication of the problem - very old Newlib version, and is consistent to previous such workaround already in the codebase.

Reply:
Then it's not just for XTensa. Perhaps the comment could be clarified by changing "for XTensa" to "for very old Newlib versions" or so?

Reply:
I hope I explained why I did like that: a) trying to, and usually go for, more general solution; b) yet explaining the motivation why ugliness like that appeared there at all.

You really think there's anything to change there (besides blacklisting Xtensa from POSIX CI tests)? Because I humbly think there's nothing to change for good, only for the purpose of bikeshedding.

Reply:
The comment says "for XTensa". While reviewing it, I wondered why it wasn't siloed away with ifdefs for that architecture only, especially to avoid unexpected changes on other architectures that are playing by the rules.

When I asked about that, you explained it's not just for XTensa. So I asked why not make the comment more accurate, which would have taken you about as much time to do as it did to write your reply.

You accused me of bikeshedding.

Have a nice day.